lib/src/functions/math.dart

```diff
@@ -133,6 +133,46 @@ final _hypot = BuiltInCallable("hypot", r"$numbers...", (arguments) {
       numeratorUnits: numeratorUnits, denominatorUnits: denominatorUnits);
 });
 
+///
+/// Exponential functions
+///
+
+final _log = BuiltInCallable("log", r"$number, $base: null", (arguments) {
+  var number = arguments[0].assertNumber("number");
+  if (arguments[1] == sassNull) {
+    number.assertNoUnits();
+    return SassNumber(math.log(number.value));
+  }
+  var base = arguments[1].assertNumber("base");
+  if (number.hasUnits) {
+    throw SassScriptException("Expected \$number to have no units.");
+  }
+  if (base.hasUnits) {
+    throw SassScriptException("Expected \$base to have no units.");
+  }
+  return SassNumber(math.log(number.value) / math.log(base.value));
+});
+
+final _pow = BuiltInCallable("pow", r"$base, $exponent", (arguments) {
+  var base = arguments[0].assertNumber("base");
+  var exponent = arguments[1].assertNumber("exponent");
+  if (base.hasUnits) {
+    throw SassScriptException("Expected \$base to have no units.");
+  }
+  if (exponent.hasUnits) {
+    throw SassScriptException("Expected \$exponent to have no units.");
+  }
+  return base.value.abs() == 1 && exponent.value.isInfinite
```

Comment:
This is a place where, rather than using Dart's `==`, you should use Sass's `fuzzyEquals()` to ensure that numbers Sass considers equivalent to `1` are treated the same as a real `1`.

Reply:
Done.